src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,169 @@
+package duke;
+
+import duke.command.Command;
+import duke.command.HelpCommand;
+import duke.command.ListCommand;
+import duke.command.QuitCommand;
+import duke.command.DoneCommand;
+import duke.command.DeleteCommand;
+import duke.command.TodoCommand;
+import duke.command.DeadlineCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.exception.EmptyDescriptionException;
+import duke.exception.InvalidCommandException;
+import duke.exception.InvalidFormatException;
+
+import java.util.Arrays;
+
+/**
+ * The Parser is a utility class that is used to parse and process
+ * String input.
+ */
+public class Parser {
+
+    /**
+     * Returns an executable Command class based on the input.
+     * If the input does not match any of the default commands,
+     * it will throw InvalidCommandException.
+     *
+     * @param input command given by the user through console.
+     * @return executable Command class.
+     * @throws InvalidCommandException
+     */
+    public static Command parse(String input) throws InvalidCommandException {
+        String[] splitted = input.split("\\s+");
+        String command = splitted[0];
+        if (command.equals("help")) {
+            return new HelpCommand();
+        } else if (command.equals("bye")) {
+            return new QuitCommand();
+        } else if (command.equals("list")) {
+            return new ListCommand();
+        } else if (command.equals("done")) {
+            return new DoneCommand();
+        } else if (command.equals("delete")) {
+            return new DeleteCommand();
+        } else if (command.equals("todo")) {
+            return new TodoCommand();
+        } else if (command.equals("deadline")) {
+            return  new DeadlineCommand();
+        } else if (command.equals("event")) {
+            return new EventCommand();
+        } else if (command.equals("find")) {
+            return new FindCommand();
+        } else {
+            throw new InvalidCommandException();
+        }
+    }
+
+    /**
+     * Captures and returns the index of a task specified by the user input.
+     *
+     * @param input command given by the user through console.
+     * @return taskId.
+     */
+    public static int parseTaskId(String input) {
+        String rawNum = input.replaceAll("[^0-9]", "");
+        int taskId = Integer.parseInt(rawNum) - 1;
+        return taskId;
+    }
+
+    /**
+     * Captures and returns the description of todo task from the user input.
+     * @param input command given by the user through console.
+     * @return description of the todo task.
+     * @throws EmptyDescriptionException
+     */
+    public static String parseTodo(String input) throws EmptyDescriptionException {

```

Comment:
It may seem like there is a lot of duplicate code among the methods below. Perhaps some of it can be extracted out.